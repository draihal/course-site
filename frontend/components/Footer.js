import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faFacebookSquare } from '@fortawesome/free-brands-svg-icons'
import { faTelegram } from '@fortawesome/free-brands-svg-icons'
import { faInstagram } from '@fortawesome/free-brands-svg-icons'
import { faVk } from '@fortawesome/free-brands-svg-icons'

import Link from 'next/link';


const Footer = () => (
    <div id='footer'>
        <footer>
            <div className='container' id='sticky-footer'>
                <div className='row footer-columns justify-content-center'>
                    <div className='footer-column col-md-2'>
                        <ul className='list-unstyled'>
                            <li><a href='#' target='_blank'>
                                <FontAwesomeIcon icon={faFacebookSquare} size='2x' />
                            </a></li>
                            <li><a href='#' target='_blank'>
                                <FontAwesomeIcon icon={faTelegram} size='2x' />
                            </a></li>
                            <li><a href='#' target='_blank'>
                                <FontAwesomeIcon icon={faInstagram} size='2x' />
                            </a></li>
                            <li><a href='#' target='_blank'>
                                <FontAwesomeIcon icon={faVk} size='2x' />
                            </a></li>
                        </ul>
                    </div>
                    <div className='footer-column col-md-2'>
                        <ul className='list-unstyled'>
                            <li><Link href='/about'><a>О нас</a></Link></li>
                            <li><a href='#'>СМИ о нас</a></li>
                            <li><a href='#'>Отзывы</a></li>
                            <li><a href='#'>Контакты</a></li>
                            <li><a href='#'>FAQ</a></li>
                        </ul>
                    </div>
                    <div className='footer-column col-md-2'>
                        <ul className='list-unstyled'>
                            <li><a href='#'>Мероприятия</a></li>
                            <li><a href='#'>Каталог курсов</a></li>
                            <li><a href='#'>Наши партнеры</a></li>
                            <li><a href='#'>Блог</a></li>
                            <li><a href='#'>Программы лояльности</a></li>
                        </ul>
                    </div>
                    <div className='footer-column col-md-3'>
                        <a href='tel:+79259999999' title='+7 925 999 99 99'>+7 925 999 99 99</a>
                        <p>По всем вопросам пишите на
                            <a href='mailto:help@test.com' target='_blank' title='help@test.com'> help@test.com</a>
                        </p>
                    </div>
                </div>
            </div>
            <div className='row justify-content-center'>
                <div className='col-md-5 text-center socket-div'>
                    <hr className='socket' />
                        © 2019-{new Date().getFullYear()} Course Site
                </div>
            </div>
        </footer>
    </div>
);

export default Footer;
