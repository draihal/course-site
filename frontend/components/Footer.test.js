import React from 'react'
import renderer from 'react-test-renderer'
import Footer from './Footer'


describe('Footer', () => {
  it('render correctly footer component', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot()
  });
});