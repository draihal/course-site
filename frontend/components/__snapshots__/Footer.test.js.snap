// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render correctly footer component 1`] = `
<div
  id="footer"
>
  <footer>
    <div
      className="container"
      id="sticky-footer"
    >
      <div
        className="row footer-columns justify-content-center"
      >
        <div
          className="footer-column col-md-2"
        >
          <ul
            className="list-unstyled"
          >
            <li>
              <a
                href="#"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-facebook-square fa-w-14 fa-2x "
                  data-icon="facebook-square"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="#"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-telegram fa-w-16 fa-2x "
                  data-icon="telegram"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 496 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm121.8 169.9l-40.7 191.8c-3 13.6-11.1 16.9-22.4 10.5l-62-45.7-29.9 28.8c-3.3 3.3-6.1 6.1-12.5 6.1l4.4-63.1 114.9-103.8c5-4.4-1.1-6.9-7.7-2.5l-142 89.4-61.2-19.1c-13.3-4.2-13.6-13.3 2.8-19.7l239.1-92.2c11.1-4 20.8 2.7 17.2 19.5z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="#"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-instagram fa-w-14 fa-2x "
                  data-icon="instagram"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="#"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-vk fa-w-18 fa-2x "
                  data-icon="vk"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M545 117.7c3.7-12.5 0-21.7-17.8-21.7h-58.9c-15 0-21.9 7.9-25.6 16.7 0 0-30 73.1-72.4 120.5-13.7 13.7-20 18.1-27.5 18.1-3.7 0-9.4-4.4-9.4-16.9V117.7c0-15-4.2-21.7-16.6-21.7h-92.6c-9.4 0-15 7-15 13.5 0 14.2 21.2 17.5 23.4 57.5v86.8c0 19-3.4 22.5-10.9 22.5-20 0-68.6-73.4-97.4-157.4-5.8-16.3-11.5-22.9-26.6-22.9H38.8c-16.8 0-20.2 7.9-20.2 16.7 0 15.6 20 93.1 93.1 195.5C160.4 378.1 229 416 291.4 416c37.5 0 42.1-8.4 42.1-22.9 0-66.8-3.4-73.1 15.4-73.1 8.7 0 23.7 4.4 58.7 38.1 40 40 46.6 57.9 69 57.9h58.9c16.8 0 25.3-8.4 20.4-25-11.2-34.9-86.9-106.7-90.3-111.5-8.7-11.2-6.2-16.2 0-26.2.1-.1 72-101.3 79.4-135.6z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <div
          className="footer-column col-md-2"
        >
          <ul
            className="list-unstyled"
          >
            <li>
              <a
                href="/about"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                О нас
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                СМИ о нас
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Отзывы
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Контакты
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                FAQ
              </a>
            </li>
          </ul>
        </div>
        <div
          className="footer-column col-md-2"
        >
          <ul
            className="list-unstyled"
          >
            <li>
              <a
                href="#"
              >
                Мероприятия
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Каталог курсов
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Наши партнеры
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Блог
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Программы лояльности
              </a>
            </li>
          </ul>
        </div>
        <div
          className="footer-column col-md-3"
        >
          <a
            href="tel:+79259999999"
            title="+7 925 999 99 99"
          >
            +7 925 999 99 99
          </a>
          <p>
            По всем вопросам пишите на
            <a
              href="mailto:help@test.com"
              target="_blank"
              title="help@test.com"
            >
               help@test.com
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      className="row justify-content-center"
    >
      <div
        className="col-md-5 text-center socket-div"
      >
        <hr
          className="socket"
        />
        © 2019-
        2020
         Course Site
      </div>
    </div>
  </footer>
</div>
`;
